### DISCUSSION TITLE
Import of PCAN-Exploter trace file (.trc)

### DISCUSSION THREAD

**Discussion by madmanguruman** [collin80/SavvyCAN#631] [Q&A] [answered]
We're trying to import a PCAN-Explorer trace file (version 1.3) into SavvyCAN V208. We are noticing the following issues:

- Timestamps are appearing as long integer values (e.g. 95726) instead of elapsed-time values (95.726)
- All CAN IDs are appearing as 0x000

Bus number, direction, DLC and payloads are appearing correctly.

Below is an excerpt of the .trc file we are dealing with. 

`;$FILEVERSION=1.3
;$STARTTIME=45043.5343670138973
;
;   Start time: 27/04/2023 12:49:29.310.0
;   Generated by PCAN-Explorer v6.6.1.2744
;-------------------------------------------------------------------------------
;   Bus  Connection   Net Connection              Protocol  Bit rate
;   2    Connection1  4x4_CANA_BDHS_500@pcan_usb  CAN       500 kbit/s
;-------------------------------------------------------------------------------
;   Message   Time    Bus  Type   ID    Reserved
;   Number    Offset  |    |      [hex] |   Data Length Code
;   |         [ms]    |    |      |     |   |    Data [hex] ...
;   |         |       |    |      |     |   |    |
;---+-- ------+------ +- --+-- ---+---- +- -+-- -+ -- -- -- -- -- -- --
     1)        95.726 2  Rx    18FAD549 -  8    E0 00 00 00 00 00 00 00
     2)       223.939 2  Rx    18FAD549 -  8    E0 00 00 00 00 00 00 00
     3)       607.716 2  Rx    18FAD549 -  8    E0 00 00 00 00 00 00 00
     4)       735.927 2  Rx    18FAD549 -  8    E0 00 00 00 00 00 00 00
     5)      1119.967 2  Rx    18FAD549 -  8    E0 00 00 00 00 00 00 00
     6)      1248.001 2  Rx    18FAD549 -  8    E0 00 00 00 00 00 00 00
     7)      1376.021 2  Rx    18FAD549 -  8    E0 00 00 00 00 00 00 00
     8)      1503.763 2  Rx    18FAD549 -  8    E0 00 00 00 00 00 00 00
     9)      1631.880 2  Rx    18FAD549 -  8    E0 00 00 00 00 00 00 00
    10)      1759.710 2  Rx    18FAD549 -  8    E0 00 00 00 00 00 00 00
    11)      1760.001 2  Rx    18FAD549 -  8    E0 00 00 00 00 00 00 00
    12)      1887.833 2  Rx    18FAD549 -  8    E0 00 00 00 00 00 00 00
    13)      1888.117 2  Rx    18FD1549 -  8    00 80 1A 01 00 FF FF FF`

Here is what we are seeing in SavvyCAN:

![image](https://user-images.githubusercontent.com/12100879/236556482-9b7ac675-136e-4de9-bb4e-6dc74d95341d.png)

Is there something we could do to preserve the CAN IDs and fix the time offsets?

**Comment by collin80** [comment, marked as answer]
There was indeed a bug in the code to load PCAN 1.3 files. This is fixed with commit 01953e675841ec9bc308320b844b55a54688dc19

You will need to use the continuous build in order to get this change or bring in the code and compile it yourself.

However, the timestamps were being correctly loaded. The program defaults to showing times in microseconds and that's what you're seeing. Go into preferences and tell it you want timestamps in seconds instead. Chances are I should think about making seconds the default. It might make more sense since I use seconds virtually always.

**Reply by madmanguruman** [reply]
Thanks for the clarification and explanation. I will give the  continuous build a try. Appreciate the support!
